### DISCUSSION TITLE
fail2ban fails to ban

### DISCUSSION THREAD

**Discussion by HedgeShot** [nextcloud/all-in-one#921] [Questions] [answered]
Hi, 

I followed the steps described on how to setup fail2ban. Unfortunately, fail2ban fails to ban... 

`fail2ban-client status nextcloud` shows:

```
Status for the jail: nextcloud
|- Filter
|  |- Currently failed:	2
|  |- Total failed:	6
|  `- File list:	/var/lib/docker/volumes/nextcloud_aio_nextcloud/_data/data/nextcloud.log
`- Actions
   |- Currently banned:	1
   |- Total banned:	1
   `- Banned IP list:	my.external.ip
```

However, I can still access cloud.my.domain from my.external.ip. I am aware that docker bypasses UFW rules so maybe it is the same with fail2ban / iptables. 

Anybody knows a workaround? If not, I believe that the fail2ban part in the AIO docs is irrelevant.

**Comment by HedgeShot** [comment, marked as answer]
ok turns out the fix was incredibly easy... `chain=DOCKER-USER`  needs to be added to `nextcloud.local`.

```
[nextcloud]
backend = auto
enabled = true
port = 80,443
protocol = tcp
filter = nextcloud
maxretry = 3
bantime = 86400
findtime = 43200
logpath = /var/lib/docker/volumes/nextcloud_aio_nextcloud/_data/data/nextcloud.log
chain=DOCKER-USER
```

**Reply by szaimen** [reply]
Hi, can you please open a PR that adds this to the fail2ban documentation in this repo's readme file?

**Reply by HedgeShot** [reply]
done

**Reply by szaimen** [reply]
Thanks!
